tests/go/go-kafka-serde/client.go

```diff
@@ -23,7 +23,7 @@ import (
 
 	log "github.com/sirupsen/logrus"
 
-	"github.com/redpanda-data/kgo-verifier/pkg/util"
+	"github.com/redpanda-data/redpanda/tools/kgo-verifier/pkg/util"
```

Comment:
I think we should move `kgo-verifier` into `tests/go` and also hoist the `go.mod` to be `tests/go/go.mod` and share between all the versions. We really should be sharing these dependencies.

I could also be convinced to do this in a followup, but let's at least move `kgo-verifier` into the `tests/go`

Reply:
I thought about this but couldn't convince myself to do it. I'm thinking about kgo-verifier as a generic tool mainly use for testing but it is also useful outside of our regular ducktape runs. I use it from cli/adhoc from time to time, etc.

Happy to move it as I only slightly lean towards `tools`. I also saw there other "testing" tools.

Single go.mod is a good argument for moving everything together. Though this won't solve in the general case all the problems as I can imagine dependencies on rpk, in theory. Other stuff might come up in the future too.

Less go mods is better so I'll probably move. Let me know if you didn't change your mind.

Reply:
Should be good now.